synapseclient/api/configuration_services.py

```diff
@@ -0,0 +1,136 @@
+"""This module is responsible for exposing access to any configuration either through
+file, environment variables, or other means.
+"""
+
+import configparser
+import functools
+from typing import Dict
+
+from synapseclient.core.constants import config_file_constants
+from synapseclient.core.pool_provider import DEFAULT_NUM_THREADS
+
+
+@functools.lru_cache()
+def get_config_file(config_path: str) -> configparser.RawConfigParser:
+    """
+    Retrieves the client configuration information.
+
+    Arguments:
+        config_path:  Path to configuration file on local file system
+
+    Returns:
+        A RawConfigParser populated with properties from the user's configuration file.
+    """
+
+    try:
+        config = configparser.RawConfigParser()
+        config.read(config_path)  # Does not fail if the file does not exist
+        return config
+    except configparser.Error as ex:
+        raise ValueError(f"Error parsing Synapse config file: {config_path}") from ex
+
+
+def get_config_section_dict(
+    section_name: str,
+    config_path: str,
+) -> Dict[str, str]:
+    """
+    Get a profile section in the configuration file with the section name.
+
+    Arguments:
+        section_name: The name of the profile section in the configuration file
+        config_path:  Path to configuration file on local file system
+
+    Returns:
+        A dictionary containing the configuration profile section content
```

Comment:
This should mention that it will return an empty dictionary if it runs into a `NoSectionError`